File: server/app/models/db/general_resource/module_response.py
from pydantic import BaseModel
from app.db.enums import AvailableCourse, AvailableDialect


class ModuleResponse(BaseModel):
    id: int
    course: AvailableCourse
    dialect: AvailableDialect | None
    unit: str
    section: str
    title: str
    number: int
    resource_id: int

File: client/types/response_models/ModuleResponse.ts
import { AvailableCourse, AvailableDialect } from "../enums";

export interface ModuleResponse {
    id: number;
    course: AvailableCourse;
    dialect: AvailableDialect | null;
    unit: string;
    section: string;
    title: string;
    number: number;
    resource_id: number
}
